.github/workflows/build-all-arduino.yml

```
@@ -0,0 +1,91 @@
+        ls -lh artifacts
+        echo "=== Failed Build Logs ==="
+        find compiled_binaries -name "compile.log" -exec sh -c 'echo "\n{}:"; cat {}' \;
+
+    - name: 📤 Upload artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: firmware-binaries
+        path: artifacts/*.bin
+
+    - name: 🏷️ Create release
+      uses: softprops/action-gh-release@v1
```

actionlint (1.7.7)

the runner of "softprops/action-gh-release@v1" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)